src/lib/table/table.md

```diff
@@ -35,60 +48,93 @@ Here's a simple column definition with the name `'userName'`. The header cell co
-one or more custom `DataSource` to capture more specific behaviors.
+To start, create a variable in your component that contains the list of the columns you want to
+render. 
 
 ```ts
-this.myDataSource = new MatTableDataSource();
-this.myDataSource.data = dataToRender;
+columnsToDisplay = ['userName', 'age'];
 ```
 
 ```html
-<mat-table [dataSource]=”myDataSource”>
-  ...
-</mat-table>
+<mat-header-row *matHeaderRowDef="columnsToDisplay"></mat-header-row>
+<mat-row *matRowDef="let myRowData; columns: columnsToDisplay"></mat-row>
 ```
 
+Note that this list of columns provided to the rows can be in any order, not necessary the order in
+which you wrote the column definitions. Also, you do not necessarily have to include every column
+that was defined in your template.
+
+This means that by changing your column list provided to the rows, you can easily re-order and
+include/exclude columns dynamically.  
+
+### Alternative ways to provide data
+
+The simplest way to provide data to your table is by passing a data array. More complex use-cases may benefit from a
+more flexible approach involving an Observable stream or by encapsulating your data source logic into a `DataSource`
+class.
+
+#### Observable stream of data arrays
+
+An alternative approach to providing data to the table is by passing an Observable stream that emits
+the data array to be rendered each time it is changed. The table will listen to this stream and 
+automatically trigger an update to the rows each time a new data array is emitted.
+
+#### DataSource
+
+Also, another approach to providing data is by implementing a `DataSource`. This is simply
+an interface that has two functions: `connect` and `disconnect`. After passing a DataSource to the
+table, the table will call `connect` to receive a stream that emits the data array that should be
+rendered. The table will call `disconnect` when the table is destroyed, which may be the right time
+to clean up any subscriptions that may have been registered during the `connect` process.
+
+The benefit to creating a `DataSource` class is that it can decouple rendering logic from your
+component and keep it encapsulated in a separate class (e.g. sorting, filtering, pagination, e.g.).
+
 ### Features
 
+The `MatTable` is focused on a single responsibility: efficiently render rows of data in a 
+performant and accessible way.
+
+You'll notice that the table itself doesn't come out of the box with a lot of features, but expects
+that the table will be included in a composition of components that fills out its features.
+
+For example, you can add sorting and pagination to the table by using MatSort and MatPaginator and
+mutating the data provided to the table according to their outputs.
+
+To simplify the use case of having a table that can sort, paginate, and filter an array of data,
+the Angular Material library comes with a `MatTableDataSource` that has already implemented
```

Comment:
I still want to rename this to `ClientArrayDataSource` (or `ClientDataSource` or `ArrayDataSource` or even `SimpleDataSource`)

My problem with the current name is that it implies that it is **the** way to provide data for a `<mat-table>`, wich may lead people to think that the table doesn't support server side sorting/pagination/filtering, or even just other kinds of filtering than search.

Reply:
Sure, I'm open to renaming. We should brainstorm offline to figure out the best name